java/core/src/main/java/com/microsoft/bond/ListBondType.java

```diff
@@ -154,7 +154,7 @@ protected final void serializeField(
             StructBondType.StructField<List<TElement>> field) throws IOException {
         this.verifySerializedNonNullableFieldIsNotSetToNull(value, field);
         int count = value.size();
-        if (count == 0 && field.isOptional()) {
+        if (!field.isDefaultNothing() && count == 0 && field.isOptional()) {
             context.writer.writeFieldOmitted(BondDataType.BT_LIST, field.getId(), field);
         } else {
```

Comment:
A field with "= nothing" should be serialized using serializeSomethingField which does that check. The StructField instance for each field knows which method to call for serialization. The same pattern holds for deserialization.